github-actions/unified-status-check/src/main.ts

```diff
@@ -67,92 +29,37 @@ async function main() {
 
       await github.repos.createCommitStatus({
         ...context.repo,
-        sha: pullRequest.commits.nodes[0].commit.oid,
+        sha: pullRequest.sha,
         context: unifiedStatusCheckName,
-        state: state.toLowerCase() as 'pending' | 'success' | 'failure',
+        state,
         description,
       });
       return;
     };
 
     /** If no status checks are present, or if the pull request is in a draft state the unified status is in a pending state. */
     const isDraftValidationResult = isDraft(pullRequest);
-    if (isDraftValidationResult.state === 'PENDING') {
+    if (isDraftValidationResult.state === 'pending') {
       await setStatus(isDraftValidationResult.state, isDraftValidationResult.description);
       return;
     }
 
-    const missedStatuses = required.filter(
-      (matcher) => !statuses.some(({name}) => name.match(matcher)),
-    );
-    if (missedStatuses.length > 0) {
-      await setStatus(
-        'PENDING',
-        `Pending ${missedStatuses.length} status(es): ${missedStatuses.join(', ')}`,
-      );
-      return;
-    }
-
-    const counts = statuses.reduce(
-      (count, {state}) => {
-        if (isPassingState(state)) {
-          count.passing += 1;
-        }
-        if (isPendingState(state)) {
-          count.pending += 1;
-        }
-        if (isFailingState(state)) {
-          count.failing += 1;
-        }
-        return count;
-      },
-      {passing: 0, failing: 0, pending: 0},
-    );
-
-    if (counts.failing > 0) {
-      await setStatus('FAILURE', `${counts.failing} expected status(es) failing`);
+    const hasRequiredStatusesResult = checkRequiredStatuses(pullRequest);
+    if (hasRequiredStatusesResult.state === 'pending') {
```

Comment:
Out of curiosity: Do you want to use `!== "success"` here? What if the required status result would technically return `failure`? It seems quite context-dependent (i.e. need to know what this function is doing)

Reply:
I am going to leave it for now as I will end up redoing this logic in a follow up so that we can get the combined result of multiple validations.

Reply:
Sounds good. Just be aware that `${statuses.failing} expected status(es) failing` is not working right now then.